src/nupic/research/temporal_memory.py

```diff
@@ -164,15 +163,15 @@ def compute(self, activeColumns, learn=True):
     segToCol = lambda segment: int(segment.segment.cell / self.cellsPerColumn)
     identity = lambda column: int(column)
 
-    for columnData in groupByN(activeColumns, identity,
+    for columnData in groupby2(activeColumns, identity,
                                self.activeSegments, segToCol,
                                self.matchingSegments, segToCol):
       (column,
        activeColumns,
        activeSegmentsOnCol,
        matchingSegmentsOnCol) = columnData
-      if len(activeColumns):
-        if len(activeSegmentsOnCol) != 0:
+      if not activeColumns is None:
```

Comment:
For these it's better to do `is not None`.

I just checked and it's in PEP-8: "Use `is not` operator rather than `not ... is`" https://www.python.org/dev/peps/pep-0008/